src/RelayClient.ts

```diff
@@ -126,15 +141,88 @@ class RelayClient extends EnvelopingEventEmitter {
 
     return BigNumber.from(gasPrice.toFixed());
   };
-  public async isSmartWalletOwner(smartWalletAddress: string, owner:string): Promise<boolean>{
-    const iForwarder = IForwarder__factory.connect(smartWalletAddress, this._provider);
-    if(await iForwarder.getOwner()  !== utils.solidityKeccak256(['address'], [owner])){
+
+  public async isSmartWalletOwner(
+    smartWalletAddress: string,
+    owner: string
+  ): Promise<boolean> {
+    const iForwarder = IForwarder__factory.connect(
+      smartWalletAddress,
+      this._provider
+    );
+    if (
+      (await iForwarder.getOwner()) !==
+      utils.solidityKeccak256(['address'], [owner])
+    ) {
       return false;
     }
 
     return true;
   }
-  
+
+  // TODO: lift the ts exception once method is used
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  async estimateTokenTransferGas(
+    requestConfig: RequestConfig,
+    envelopingRequest: EnvelopingRequest,
+    internalTransactionEstimateCorrection = INTERNAL_TRANSACTION_ESTIMATE_CORRECTION,
+    estimatedGasCorrectionFactor = 1
+  ): Promise<BigNumber> {
+    const tokenContract = envelopingRequest.request.tokenContract.toString();
+
+    if (
+      tokenContract === constants.AddressZero ||
+      envelopingRequest.request.tokenAmount.toString() === '0'
+    ) {
+      return ethers.constants.Zero;
+    }
+
+    let tokenOrigin: string | undefined;
+
+    if (requestConfig.isSmartWalletDeploy) {
+      tokenOrigin = requestConfig?.smartWalletAddress;
+
+      // If it is a deploy and tokenGas was not defined, then the smartwallet address
+      // is required to estimate the token gas. This value should be calculated prior to
+      // the call to this function
+      if (tokenOrigin === constants.AddressZero) {
+        throw Error(MISSING_SMART_WALLET_ADDRESS);
+      }
+    } else {
+      tokenOrigin = envelopingRequest.relayData.callForwarder.toString();
+
+      if (tokenOrigin === constants.AddressZero) {
+        throw Error(MISSING_CALL_FORWARDER);
+      }
+    }
+
+    const erc20 = IERC20__factory.connect(tokenContract, this._provider);
+    const gasCost = await erc20.estimateGas.transfer(
+      tokenContract,
+      envelopingRequest.request.tokenAmount,
+      { from: tokenOrigin }
+    );
+
+    const internalCallCost = gasCost.gt(internalTransactionEstimateCorrection)
+      ? gasCost.sub(internalTransactionEstimateCorrection)
+      : gasCost;
```

Comment:
If we agree on having a [single function that applies this logic](https://github.com/rsksmart/rif-relay-client/pull/55/files#r1048742211), we could call it here also.